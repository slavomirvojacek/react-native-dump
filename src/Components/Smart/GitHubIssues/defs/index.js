const userDef = (id = '') => ({id})

const reactionDef = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDef()
  }

  return {...def, ...override}
}

const issueDef = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

export {
  userDef, reactionDef, issueDef
}
