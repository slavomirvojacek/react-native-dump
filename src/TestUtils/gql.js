const assertQueryOrMutation = (QoM) => {
  test('matches Query or Mutation', () => {
    expect(QoM).toMatchSnapshot()
  })
}

const assertGql = (Component) => {
  const { gql } = Component
  expect(gql instanceof Object).toEqual(true)
  Object
    .keys(gql)
    .forEach(k => assertQueryOrMutation(gql[k]))
}

export { assertGql }
