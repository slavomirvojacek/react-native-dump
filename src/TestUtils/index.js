export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
