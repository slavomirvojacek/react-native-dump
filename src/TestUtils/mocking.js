// Sometimes, we might want to test internal state changes without having to render.
// This overrides the setState method to only update the state, without interacting with
// React's internals such as the scheduler.
const mockSetState = (Component) => {
  return class extends Component {
    setState (f) {
      if (typeof f === 'function') {
        this.state = {...this.state, ...f(this.state)}
      } else {
        this.state = {...this.state, ...f}
      }
    }
  }
}

export { mockSetState }
