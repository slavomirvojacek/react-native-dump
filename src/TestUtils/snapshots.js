import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const assertSnapshots = (Component, props, desc, store = null) => {
  let getComponent = p => null

  if (store !== null) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

export { assertSnapshots }
