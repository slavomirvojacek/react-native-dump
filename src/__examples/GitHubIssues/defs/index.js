const userDef = (id = '') => ({id})

const reactionDef = ({
                       id = '',
                       content = '',
                       user = userDef()
                     }) => ({
  id, content, user
})

const issueDef = ({
                    id = '',
                    title = '',
                    url = '',
                    bodyText = '',
                    reactions = []
                  }) => ({
  id, title, url, bodyText, reactions
})

export {
  userDef, reactionDef, issueDef
}
