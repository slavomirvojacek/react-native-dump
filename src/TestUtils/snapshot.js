import React from 'react'
import renderer from 'react-test-renderer'

import { mockSetState } from './mocking'

const assertSnapshot = (C, config) => {
  const {props, state, desc, throws} = config

  if (throws) {
    test(desc, () => {
      expect(() => {
        if (C.prototype.render) {
          new C(props)
        } else {
          C(props)
        }
      }).toThrow(throws)
    })
  } else {
    let Component
    let RenderedComponent

    if (C.prototype.render) {
      const CWithMockState = mockSetState(C)
      Component = new CWithMockState(props)

      if (state) {
        Component.setState(state)
      }

      RenderedComponent = Component.render()
    } else {
      Component = C(props)
      RenderedComponent = Component
    }

    test(desc, () => {
      const tree = renderer
        .create(RenderedComponent)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
}

const assertSnapshots = (Component, configs) => {
  configs.forEach(config => assertSnapshot(Component, config))
  // if (Component.gql) assertGql(Component)
}

export { assertSnapshots }
