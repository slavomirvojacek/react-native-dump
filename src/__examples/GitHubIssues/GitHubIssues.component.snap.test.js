import React from 'react'
import { Some } from '@threestup/monads'

import { issueDef } from './defs'
import GitHubIssuesComponent from './GitHubIssues.component'
import Config from './GitHubIssues.component.config'

import { assertSnapshots } from '../../TestUtils/snapshot'

jest
  .mock('./lib/Components/Issues/Issues.component', () => 'Issues')

describe('GitHubIssues View Snapshots', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const configs = [
    {
      props: Config.defaultProps,
      desc: 'renders correctly'
    },
    {
      props: {
        ...Config.defaultProps,
        issuesLoading: true,
        issues: Some([issueDef()]),
        error: Some('Oops!')
      },
      desc: 'renders correctly when issuesLoading `true`, regardless of what else is present'
    },
    {
      props: {
        ...Config.defaultProps,
        issuesLoading: false,
        issues: Some([issueDef()]),
      },
      desc: 'renders correctly when issuesLoading `false` and issues present'
    },
    {
      props: {
        ...Config.defaultProps,
        issuesLoading: false,
        issues: Some([]),
      },
      desc: 'renders correctly when issuesLoading `false` and issues are empty array'
    },
    {
      props: {
        ...Config.defaultProps,
        issuesLoading: false,
        error: Some('Oops!'),
      },
      desc: 'renders correctly when issuesLoading `false` and error present'
    },
    {
      props: {
        ...Config.defaultProps,
        issuesLoading: false,
        error: Some('Oops!'),
        issues: Some([this.issueDef]),
      },
      desc: 'renders correctly when error present, regardless of whether issues are present'
    }
  ]

  assertSnapshots(GitHubIssuesComponent, configs)
})
