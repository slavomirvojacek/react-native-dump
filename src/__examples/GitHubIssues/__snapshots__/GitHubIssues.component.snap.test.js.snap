// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubIssues View Snapshots renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Cannot get Issues for some reason
  </Text>
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when \`error\` present, regardless of what else is present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops!
  </Text>
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when \`error\` present, regardless of whether issues are present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops!
  </Text>
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when error present, regardless of whether issues are present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops!
  </Text>
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when issuesLoading \`false\` and \`error\` present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops!
  </Text>
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when issuesLoading \`false\` and error present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops!
  </Text>
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when issuesLoading \`false\` and issues are empty array 1`] = `
<View>
  <Issues
    issues={Array []}
  />
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when issuesLoading \`false\` and issues empty array 1`] = `
<View>
  <Issues
    issues={Array []}
  />
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when issuesLoading \`false\` and issues present 1`] = `
<View>
  <Issues
    issues={
      Array [
        Object {
          "bodyText": "",
          "id": "",
          "reactions": Array [],
          "title": "",
          "url": "",
        },
      ]
    }
  />
</View>
`;

exports[`GitHubIssues View Snapshots renders correctly when issuesLoading \`true\`, regardless of what else is present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Issues are currently loading...
  </Text>
</View>
`;
